### PR TITLE
Lengine

### PR DESCRIPTION
**Pull request by timocafe**
- support intel compiler for LE

### CODE REVIEW

**Comment by tristan0x** [comment]
There are issues at link time:
```
CMakeFiles/unit_example_simulations.dir/learnengine.cpp.o: In function `void lengine::helper_mkl<double>(void*, lengine::block<double, lengine::align, lengine::zlib>::pointer, int, unsigned long, unsigned long)':
learnengine.cpp:(.text._ZN7lengine10helper_mklIdEEvPvNS_5blockIT_NS_5alignENS_4zlibEE7pointerEimm[_ZN7lengine10helper_mklIdEEvPvNS_5blockIT_NS_5alignENS_4zlibEE7pointerEimm]+0x15): undefined reference to `vdRngUniform'
CMakeFiles/unit_example_simulations.dir/learnengine.cpp.o: In function `void lengine::helper_mkl<float>(void*, lengine::block<float, lengine::align, lengine::zlib>::pointer, int, unsigned long, unsigned long)':
learnengine.cpp:(.text._ZN7lengine10helper_mklIfEEvPvNS_5blockIT_NS_5alignENS_4zlibEE7pointerEimm[_ZN7lengine10helper_mklIfEEvPvNS_5blockIT_NS_5alignENS_4zlibEE7pointerEimm]+0x15): undefined reference to `vsRngUniform'
../../../lengine/libengine.so: undefined reference to `vslNewStream'
```

**Comment by tristan0x** [comment]
retest this please

File: bbp/default.nix [hunk 1]
```diff
@@ -443,6 +443,7 @@ let
 
         learningengine = callPackage ./hpc/learningengine {
             stdenv = stdenvIntelfSupported;
+            blas = intelMKLIfSupported;
```
Comment (tristan0x) [inline bbp/default.nix]: I think this addition is unnecessary

File: bbp/hpc/learningengine/default.nix [hunk 2]
```diff
@@ -7,6 +7,7 @@
 , hdf5
 , highfive
 , tbb
+, blas
```
Comment (tristan0x) [inline bbp/hpc/learningengine/default.nix]: I think this addition is unnecessary

File: bbp/hpc/learningengine/default.nix [hunk 3]
```diff
@@ -34,6 +35,7 @@ stdenv.mkDerivation rec {
                     pkgconfig
                     syntool
                     hdf5
+                    blas
```
Comment (tristan0x) [inline bbp/hpc/learningengine/default.nix]: I think this addition is unnecessary

Comment (adevress) [reply bbp/hpc/learningengine/default.nix]: No, it is for future MKL support :)

**Comment by tristan0x** [comment]
There are still unnecessary references to the `blas` package in both `bbp/default.nix` and `bbp/hpc/learningengine/default.nix`. It `blas` is not part of the `buildInputs`, I suggest we also remove them.

**Pull request merged**